### ISSUE TITLE
What's the functionality of applying EqualLinear in psp_encoders 

### ISSUE THREAD

**Issue by XD666** [eladrich/pixel2style2pixel#23]
For psp_encoders.py L148 
x = x.view(-1, 512) It's supposed to get the latent from here but it seems you do an EqualLinear further in the next step:
x = self.linear(x)
What's the function here?  I see all these Encoders have applied for EqualLinear. 
Any experimental differences if get rid of this layer? I didn't find them in the paper so post here. 
Looking forward to hearing your clarifying! Appreciate it!

**Comment by eladrich** [comment]
Hi @XD666, good question! Let me try and explain.
The EqualLinear layer is originally defined as part of  the StyleGAN model (see [here)](https://github.com/eladrich/pixel2style2pixel/blob/master/models/stylegan2/model.py#L128), basically It's a fully connected layer, followed by PReLU, where a fixed scaling of weights is set during training, which presumably stabilizes training (for more see `Section 4.1 Equalized Learning Rate` in `Progressive Growing of GANs for Improved Quality, Stability, and Variation`)

As this layer was originally used to transform the `z` latent in StyleGAN into `w` through the mapping network, we thought it would make sense to apply it as a last transformation on the output of our encoder. It is not mentioned in the paper as we thought it would be too technical there (which is why we belive it's important to release code!). As to its effectiveness, we don't have complete ablation studies there, but you can give it a try yourself and let us know :-)

**Comment by XD666** [comment]
Thank you very much for the explanation! @eladrich  It makes a lot of sense to apply it to follow StyleGAN protocol. I didn't notice these details and eliminated this layer and didn't get the right way where the outputs from Stylegan even not like a face. Seems this layer makes more stable and easy to convergency. I believe these tips are pretty useful.
